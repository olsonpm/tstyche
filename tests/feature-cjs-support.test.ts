import { expect, test } from "@jest/globals";
import { normalizeOutput } from "./__utils__/normalizeOutput.js";
import { spawnTyche } from "./__utils__/spawnTyche.js";

const fixture = "feature-cjs-support";

test("supports CJS projects written CJS syntax", async () => {
  const { exitCode, stderr, stdout } = await spawnTyche(fixture, ["cjs-syntax"]);

  expect(normalizeOutput(stdout)).toMatchSnapshot("stdout");
  expect(normalizeOutput(stderr)).toMatchSnapshot("stderr");

  expect(exitCode).toBe(0);
});

test("supports CJS projects written in ESM syntax", async () => {
  const { exitCode, stderr, stdout } = await spawnTyche(fixture, ["esm-syntax"]);

  expect(normalizeOutput(stdout)).toMatchSnapshot("stdout");
  expect(normalizeOutput(stderr)).toMatchSnapshot("stderr");

  expect(exitCode).toBe(0);
});
