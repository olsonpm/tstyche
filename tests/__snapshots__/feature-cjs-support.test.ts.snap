// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`supports CJS projects written CJS syntax: stderr 1`] = `""`;

exports[`supports CJS projects written CJS syntax: stdout 1`] = `
"uses TypeScript <<version>> with ./tsconfig.json

pass ./__typetests__/cjs-syntax.tst.cts

Targets:    1 passed, 1 total
Test files: 1 passed, 1 total
Duration:   <<timestamp>>

Ran test files matching 'cjs-syntax'.
"
`;

exports[`supports CJS projects written in ESM syntax: stderr 1`] = `""`;

exports[`supports CJS projects written in ESM syntax: stdout 1`] = `
"uses TypeScript <<version>> with ./tsconfig.json

pass ./__typetests__/esm-syntax.tst.ts
  + is supported?

Targets:    1 passed, 1 total
Test files: 1 passed, 1 total
Tests:      1 passed, 1 total
Assertions: 1 passed, 1 total
Duration:   <<timestamp>>

Ran test files matching 'esm-syntax'.
"
`;
